### ISSUE TITLE
unkown on pokestops after a while

### ISSUE THREAD

**Issue by kazyka** [OpenPoGo/OpenPoGoBot#60]
After a while I get this `Now at Pokestop: Unknown - Spinning...`

**Comment by sakulstra** [comment]
Was also a bug in the original repo: https://github.com/PokemonGoF/PokemonGo-Bot/issues/794

And should be fixed via: https://github.com/PokemonGoF/PokemonGo-Bot/commit/74bbe3f7203cd180a43cc8ae52ae13dbfc34830e and https://github.com/PokemonGoF/PokemonGo-Bot/commit/1312ca4aa2b6ea800ac4637d3964ae1a9ab6668b

**Comment by kazyka** [comment]
Will try to pull their newest and see. Guess it is not fixed in OpenPoGoBot?

**Comment by sakulstra** [comment]
Think it's not fixed in OpenPoGoBot nor in latest F bot master - only in dev which has a lot of bugs.

**Comment by wchill** [comment]
Yeah, currently we are refactoring to make the code more manageable. However, I'll look into cherrypicking those commits.

**Comment by wchill** [comment]
@sakulstra @kazyka Please try this branch.

https://github.com/OpenPoGo/OpenPoGoBot/tree/wchill/fix-token-refresh

**Comment by sakulstra** [comment]
@wchill not working

**Comment by wchill** [comment]
Currently testing fix.

**Comment by epicblood** [comment]
@wchill I had a similar issue, however instead of "Unkown" it would crash with
 `AttributeError: 'NoneType' object has no attribute 'encode'`

I managed to hack in a fix to stop it from crashing, and seems to be working:
seen_fort_worker.py:49 `fort_name = fort_details.get("name").encode("utf8", "replace") if fort_details.get("name") else fort_details.get("name")`

**Comment by meyer9** [comment]
epicblood, if you have a fix, submit a PR please.

**Comment by meyer9** [comment]
Fixed in #66

**Comment by sakulstra** [comment]
I think this can't be the fix... The error occured because the session token was timing out.

**Comment by kazyka** [comment]
It stops showing unkown if you stop the bot and the start it again. Maybe the workaround until a fix should be to make the bot stop and then begin at the last known coords?

**Comment by wchill** [comment]
Not fixed.

**Comment by tehp** [comment]
Should note, it happened to me after running for 3.5 hours. Same with @epicblood

**Comment by dbmurphy** [comment]
I have been playing with this and seem to have a good solution in place. Also since a re-auth is happening here, I also re-pulled player stats.

Please note in this case I change the retry to happen long before the session would expire but normally it would  refresh then the expire time is within 30 seconds.

```
2016-07-28 11:21:26] [#] Walking from (X,Y) to (Z,A) for approx. 25.00 seconds
[2016-07-28 11:22:01] [#] Remaining Session Time: 1740.37476397
[2016-07-28 11:22:12] [#] Finished walking
[2016-07-28 11:22:12] [#] Arrived at Pokestop
[2016-07-28 11:22:14] [#] Now at Pokestop: HIDDEN NAME
[2016-07-28 11:22:14] Spinning...
[2016-07-28 11:22:18] [#] Found fort 0000000000000000000000000000000.00 at distance 0.04km
[2016-07-28 11:22:18] [#] Need to move closer to Pokestop
[2016-07-28 11:22:18] [#] Walking from (X,Y) to (Z,A) for approx. 13.00 seconds
[2016-07-28 11:22:43] [X] Session stale, re-logging in
[2016-07-28 11:22:43] [#] Attempting login to Pokemon Go.
[2016-07-28 11:23:14] [X] Login Error, server busy
[2016-07-28 11:23:14] [X] Waiting X seconds to try again
[2016-07-28 11:23:26] [+] Login to Pokemon Go successful.
[2016-07-28 11:23:27] [#]
[2016-07-28 11:23:27] [#] Username: XXXXX
[2016-07-28 11:23:27] [#] Account Creation: 2016-07-XX XX:XX:XX.000000
[2016-07-28 11:23:27] [#] Bag Storage: 650/650
[2016-07-28 11:23:28] [#] Pokemon Storage: 193/250
[2016-07-28 11:23:28] [#] Stardust: XXXX
[2016-07-28 11:23:28] [#] Pokecoins: XX
[2016-07-28 11:23:28] [#] PokeBalls: 186
[2016-07-28 11:23:28] [#] GreatBalls: 13
[2016-07-28 11:23:28] [#] UltraBalls: 16
[2016-07-28 11:23:28] [#] Remaining Session Time: 1797.80011606
[2016-07-28 11:23:28] [#] Finished walking
[2016-07-28 11:23:28] [#] Arrived at Pokestop
[2016-07-28 11:23:31] [#] Now at Pokestop: Hidden Name
[2016-07-28 11:23:31] Spinning...
[2016-07-28 11:23:37] [#] Found fort 0000000000000000000000000000000.00 at distance 0.05km
[2016-07-28 11:23:37] [#] Need to move closer to Pokestop
[2016-07-28 11:23:37] [#] Walking from (X,Y) to (Z,A) for approx. 22.00 seconds
^C[2016-07-28 11:23:41] [x] Exiting PokemonGo Bot
```

Does this look like a good solution if so I can make a new pull as #61 had some buggery in multiple places.

**Comment by wchill** [comment]
Closed by #112.
